### ISSUE TITLE
[FEATURE] NumberFormat to disallow leading zeroes

### ISSUE THREAD

**Issue by ijl** [Alexhuszagh/rust-lexical#35]
## Problem

Some languages disallow leading zeros in integers (but not floats). For example, a Python REPL:

```
>>> 012
  File "<stdin>", line 1
SyntaxError: leading zeros in decimal integer literals are not permitted; use an 0o prefix for octal integers
>>> -012
  File "<stdin>", line 1
SyntaxError: leading zeros in decimal integer literals are not permitted; use an 0o prefix for octal integers
>>> 012.0
12.0
```

JavaScript/JSON has the same behavior.

## Solution

I don't see an existing format that applies. If a new format is added, it would pass this test:

```
diff --git a/lexical-core/src/atoi/api.rs b/lexical-core/src/atoi/api.rs
index 022214f..1196fd8 100644
--- a/lexical-core/src/atoi/api.rs
+++ b/lexical-core/src/atoi/api.rs
@@ -348,6 +348,14 @@ mod tests {
         assert!(i32::from_lexical_format(b"31_", format).is_err());
     }

+    #[test]
+    #[cfg(feature = "format")]
+    fn i32_leading_zero() {
+        let format = NumberFormat::INTEGER_NO_LEADING_ZERO;
+        assert!(i32::from_lexical_format(b"012", format).is_err());
+        assert!(i32::from_lexical_format(b"-012", format).is_err());
+    }
+
     #[cfg(feature = "std")]
     proptest! {
         #[test]
```

This could be applied to at least `NumberFormat::PYTHON_LITERAL` and `NumberFormat::JSON`.

**Comment by Alexhuszagh** [comment]
Thanks, there so many different features I complete forgot many languages disallow leading 0s to prevent confusion with octal literals in C. I'll start working on this shortly,

**Comment by ijl** [comment]
@Alexhuszagh ok, thank you for spending time on it.

**Comment by Alexhuszagh** [comment]
@ijl There will need to be at least 2 flags for this feature, for both integers and floating point numbers.

For example, in Python3:

```python
>>> 001
SyntaxError: invalid token
>>> 001.0
1.0
```

In Python2:

```python
>>> 001
1
>>> 001.0
1.0
```

**Comment by Alexhuszagh** [comment]
This should be fixed by the latest commit, 147d0e0d9e6c95f8895c33a05ed5b1f135b0c74e.

**Comment by Alexhuszagh** [comment]
Live as of the latest releases, thank you for reporting this.
